crates/fuel-streams-publisher/src/inputs.rs

```diff
@@ -20,239 +25,129 @@ use fuel_streams_core::{
+                                .boxed(),
+                            InputsContractSubject::WILDCARD,
+                        ),
+                        (
+                            InputsByIdSubject::new()
+                                .with_id_kind(Some(IdentifierKind::ContractID))
+                                .with_id_value(Some(contract_id.into()))
+                                .boxed(),
+                            InputsByIdSubject::WILDCARD,
+                        ),
+                    ]
+                }
+                Input::CoinSigned(CoinSigned {
+                    owner, asset_id, ..
+                })
+                | Input::CoinPredicate(CoinPredicate {
+                    owner, asset_id, ..
+                }) => {
+                    vec![
+                        (
+                            InputsCoinSubject::new()
+                                .with_tx_id(Some(tx_id.into()))
+                                .with_index(Some(index))
+                                .with_owner(Some(owner.into()))
+                                .with_asset_id(Some(asset_id.into()))
+                                .boxed(),
+                            InputsCoinSubject::WILDCARD,
+                        ),
+                        (
+                            InputsByIdSubject::new()
+                                .with_id_kind(Some(IdentifierKind::AssetID))
+                                .with_id_value(Some((*asset_id).into()))
+                                .boxed(),
+                            InputsByIdSubject::WILDCARD,
+                        ),
+                    ]
+                }
+                Input::MessageCoinSigned(MessageCoinSigned {
+                    sender,
+                    recipient,
+                    ..
+                })
+                | Input::MessageCoinPredicate(MessageCoinPredicate {
+                    sender,
+                    recipient,
+                    ..
+                })
+                | Input::MessageDataSigned(MessageDataSigned {
+                    sender,
+                    recipient,
+                    ..
+                })
+                | Input::MessageDataPredicate(MessageDataPredicate {
+                    sender,
+                    recipient,
+                    ..
+                }) => {
+                    vec![
+                        (
+                            InputsMessageSubject::new()
+                                .with_tx_id(Some(tx_id.into()))
+                                .with_index(Some(index))
+                                .with_sender(Some(sender.into()))
+                                .with_recipient(Some(recipient.into()))
+                                .boxed(),
+                            InputsMessageSubject::WILDCARD,
+                        ),
+                        (
+                            InputsByIdSubject::new()
```

Comment:
I guess we can publish two times here, one for `sender` and another for `recipient`, what do you think?